specs/vm/instruction_set.md

```diff
@@ -1785,3 +1786,101 @@ Panic if:
 - `$fp->$fp == 0` (if parent context is external)
 
 Set `$rA` to `$fp->$fp` (i.e. `$rA` will point to the previous call frame's contract ID).
+
+### GTF: Get transaction fields
+
+|             |                         |
+|-------------|-------------------------|
+| Description | Get transaction fields. |
+| Operation   | Varies (see below).     |
+| Syntax      | `gtf $rA, $rB, imm`     |
+| Encoding    | `0x00 rA rB i i`        |
+| Notes       |                         |
+
+Get [fields from the transaction](../protocol/tx_format.md#transaction).
+
+| name                                      | `imm`   | set `$rA` to                                     |
+|-------------------------------------------|---------|--------------------------------------------------|
+| `GTF_TYPE`                                | `0x001` | `tx.type`                                        |
```

Comment:
Should we consider using imm values that are easily maskable based on their compatibility with the tx?

eg. all `GTF_SCRIPT_*` are compatible with a mask of `0x00ff`, all `GTF_CREATE_*` are under a mask of `0xff00`..

I noticed there is some degree of domain separation between the input, output and witness values, but there's not much room to grow if we add a new field for script transactions.